src/arrayMethodSort.js

```diff
@@ -5,7 +5,19 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    const compareFunc = typeof compareFunction === 'function'
+      ? compareFunction
+      : (a, b) => String(a) < String(b) ? -1 : String(a) > String(b) ? 1 : 0;
```

Comment:
Would be great to move this function to a separate constant and assign it as a default value in arguments 
```
function(compareFunction = defaultCompare) { ...